lib/presentation/crowdactions/crowdaction_form/sections/commitments_section/assign_commitments/commitment_template/commitment_template_widgets/commitment_list_pages.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:collaction_cms/presentation/crowdactions/crowdaction_form/sections/commitments_section/assign_commitments/commitment_template/commitment_template_widgets/dummy_data.dart';
+import 'package:collaction_cms/presentation/crowdactions/crowdaction_form/sections/commitments_section/assign_commitments/commitment_template/commitment_template_widgets/expandable_card_list.dart';
+import 'package:collaction_cms/presentation/crowdactions/crowdaction_form/sections/commitments_section/assign_commitments/commitment_template/commitment_template_widgets/previousNextButton.dart';
+import 'package:collection/collection.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../../../../theme/constants.dart';
+
+class CommitmentListPages extends StatefulWidget {
+  final double fullWidth;
+  const CommitmentListPages({super.key, required this.fullWidth});
+
+  @override
+  State<CommitmentListPages> createState() => _CommitmentListPagesState();
+}
+
+class _CommitmentListPagesState extends State<CommitmentListPages> {
+  final PageController controller = PageController();
+  int _currentPage = 0;
+  List<Widget> pages = [];
+  static const _kDuration = Duration(milliseconds: 300);
+  static const _kCurve = Curves.ease;
+
+  @override
+  void dispose() {
+    controller.dispose();
+    super.dispose();
+  }
+
+  @override
+  void initState() {
+    buildPages(dummyData);
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        Row(
+          crossAxisAlignment: CrossAxisAlignment.center,
+          children: [
+            const SelectableText(
+              "Commitments",
+              style: CollactionTextStyles.captionStyleBold,
+            ),
+            Container(
+              margin: const EdgeInsets.only(left: 13),
+              alignment: Alignment.center,
+              decoration: BoxDecoration(
+                color: Colors.white,
```

Comment:
We already have a widget for the commitments list counter, which might help with this issue because it has a border. It is located here: `shared/extra/list_counter.dart`.

Reply:
done